test/e2e/cluster_samples_operator_test.go

```diff
@@ -33,24 +48,383 @@ func TestClusterSamplesOperator(t *testing.T) {
+	if len(isl.Items) > 0 {
+		t.Errorf("still imagestreams in openshift namespace %#v", isl.Items)
+	}
+	tl := &templatev1.TemplateList{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "TemplateList",
+			APIVersion: templatev1.SchemeGroupVersion.String(),
+		},
+	}
+	sdk.List("openshift", tl, sdk.WithListOptions(&metav1.ListOptions{}))
+	if len(tl.Items) > 0 {
+		t.Errorf("still imagestreams in openshift namespace %#v", tl.Items)
+	}
 
-	istag := &imageapiv1.ImageStreamTag{
+	sr = verifyOperatorUp(t)
+	sr.Spec.ManagementState = operatorsv1alpha1api.Managed
+	err = sdk.Update(sr)
+	if err != nil {
+		t.Errorf("error updating samplesresource %v", err)
+	}
+
+	err = verifyConditionsCompleteSamplesAdded(sr)
+	if err != nil {
+		t.Errorf("samples not re-established when set to managed %#v", sr)
+	}
+
+	validateContent(t, &now)
+
+	sr = verifyOperatorUp(t)
+	sr.Spec.ManagementState = operatorsv1alpha1api.Unmanaged
+	err = sdk.Update(sr)
+	if err != nil {
+		t.Errorf("error updating samplesresource %v", err)
+	}
+
+	is := &imageapiv1.ImageStream{
 		TypeMeta: metav1.TypeMeta{
-			Kind:       "ImageStreamTag",
+			Kind:       "ImageStream",
 			APIVersion: imageapiv1.SchemeGroupVersion.String(),
 		},
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      "ruby:2.4",
+			Name:      "jenkins",
 			Namespace: "openshift",
 		},
 	}
-	err = wait.PollImmediate(1*time.Second, 10*time.Minute, func() (bool, error) {
-		if err := sdk.Get(istag); err != nil {
-			return false, nil
+	err = sdk.Delete(is, sdk.WithDeleteOptions(&metav1.DeleteOptions{}))
+	if err != nil {
+		t.Errorf("error deleting jenkins imagestream %v", err)
+	}
+	// make sure jenkins imagestream does not appear while unmanaged
+	// first, wait sufficiently to make sure delete has gone though
+	err = wait.PollImmediate(1*time.Second, 10*time.Second, func() (bool, error) {
+		err := sdk.Get(is)
+		if kerrors.IsNotFound(err) {
+			return true, nil
 		}
-		return true, nil
+		return false, nil
+	})
+	if err != nil {
+		t.Errorf("delete did not occur %v", err)
+	}
+	// now make sure it has not been recreated
+	err = wait.PollImmediate(1*time.Second, 30*time.Second, func() (bool, error) {
+		err := sdk.Get(is)
+		if err == nil {
+			return false, fmt.Errorf("jenkins imagestream recreated")
+		}
+		return false, nil
+	})
+	if strings.Contains(err.Error(), "jenkins imagestream recreated") {
+		t.Fatalf("%v", err)
+	}
+
+	// now switch back to default managed for any subsequent tests
+	// and confirm all the default samples content, including the jenkins
+	// imagestream, exists
+	sr = verifyOperatorUp(t)
+	sr.Spec.ManagementState = operatorsv1alpha1api.Managed
+	err = sdk.Update(sr)
+	if err != nil {
+		t.Errorf("error updating samplesresource %v", err)
+	}
+
+	// wait for it to get into pending
+	err = wait.PollImmediate(1*time.Second, 30*time.Second, func() (bool, error) {
+		err := sdk.Get(sr)
+		if err != nil {
+			return false, err
+		}
+		if sr.ConditionTrue(samplesapi.ImageChangesInProgress) {
```

Comment:
seems likely you could miss this condition if the imagestream gets imported fast enough. Seems likely to be a source of flakes.

Reply:
Good catch.  I need to establish a timestamp before setting managed to true, and allow for either in progress being  true, or the last update time on the condition being greater than the pre-managed timestamp.